packages/vscode-extension/src/debug/runIconHandler.ts

```diff
@@ -22,6 +22,7 @@ export async function selectAndDebug(args?: any[]): Promise<Result<null, FxError
 }
 
 export function registerRunIcon(): void {
+  ext.context.subscriptions.push(vscode.window.onDidChangeActiveTextEditor(enableRunIcon));
```

Comment:
`isValidProject` is a bit heavy (contains I/O operation) so avoid to call it frequently.
How about only validate project when `onDidChangeWorkspaceFolders` and `onDidSaveTextDocument` of any *.fx/* file?

Reply:
Good idea, All the condition has been changed.

Reply:
`isValidProject` is required for correctly checking the whole project status, but just avoid calling it frequently.
Some options or you can add more:
1. consider `vscode.workspace.createFileSystemWatcher` to catch *.fx* changes, even out of vscode
2. set a timer to call `isValidProject` and save current status, and `onDidXxx` just refer the status flag

Reply:
According to the [document](https://code.visualstudio.com/api/references/vscode-api), the FileSystemWatcher cannot track the folder change.
> Note that only files within the current workspace folders can be watched. Note that when watching for file changes such as '*/.js', notifications will not be sent when a parent folder is moved or deleted (this is a known limitation of the current implementation and may change in the future).

Reply:
Add a cache. If the editor changed in 5s, the project status will not be checked.